### PR TITLE
[integration_test] upgrade androidx test to 1.4.0

### PR DESCRIPTION
**Pull request by micaelcid**
I just finished migrating a project to Android 13 that runs `integration_test` with Firebase Test Lab. In this PR, I am changing the exact dependencies I needed to change to make our application able to run integration tests targeting Android 13. (the project is currently depending on my fork). 

> This PR is being reopened from https://github.com/flutter/flutter/pull/109547, adding the pipeline fixes

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

### CODE REVIEW

**Comment by micaelcid** [comment]
I'm reopening this as a Draft until I solve all CI failures that appeared after merging https://github.com/flutter/flutter/pull/109547

The failures can be found here: https://github.com/flutter/flutter/runs/11827814804

**Comment by micaelcid** [comment]
Hey, @ricardoamador. My last PR was reverted, but I guess I solved the other issues now. Would you mind reviewing this again?

**Comment by ricardoamador** [comment]
Yes, will do.

**Comment by ricardoamador** [comment]
@micaelcid if you are stuck with IOS you can always try the flutter hackers-ios channel on discord and ask for some help there.

**Comment by micaelcid** [comment]
Thanks for letting me know! I'll do that as soon as I've got spare time.

**Comment by ricardoamador** [comment]
I can take a look.

**Review by ricardoamador — APPROVED** [review]
LGTM

**Comment by ricardoamador** [comment]
You need one more review @micaelcid.

**Comment by micaelcid** [comment]
Hey, @reidbaker. Would you mind reviewing this PR? I'm tagging you because you were one of the approvers of the one that got reverted :)

**Comment by micaelcid** [comment]
By the way, is there a specific process for requesting a pull request review in this repository? I usually ask for recent contributors, but I am not sure if I should tag specific teams or if there's a guideline for this.

File: dev/benchmarks/complex_layout/android/gradle/wrapper/gradle-wrapper.properties
Comment (reidbaker) [inline dev/benchmarks/complex_layout/android/gradle/wrapper/gradle-wrapper.properties:3]: I dont think we should downgrade the version of gradle. Is this required or your change?

**Review by reidbaker — CHANGES_REQUESTED** [review]

**Comment by reidbaker** [comment]
> Hey, @reidbaker. I just solved your requested changes and the pipeline checks have passed. Do you mind reviewing again?

Yep I will do my best to get to it today if not then by EOD Fri.

**Review by reidbaker — APPROVED** [review]

**Comment by micaelcid** [comment]
Are we aware of flaky tests in the GH Actions pipeline? While coming up with this PR, sporadically a few tests would fail and get solved after a rebase. For example, [these checks](https://github.com/flutter/flutter/pull/122437/checks?check_run_id=12400692862) fail intermittently, but they were previously passing.

**Comment by micaelcid** [comment]
Can you guys @ricardoamador or @reidbaker add the auto-submit label?

**Comment by micaelcid** [comment]
Hey, @goderbauer do you mind adding the auto submit label?

**Pull request merged**
